ambry-protocol/src/main/java/com/github/ambry/protocol/UndeleteResponse.java

```diff
@@ -62,6 +62,18 @@ public UndeleteResponse(int correlationId, String clientId, short lifeVersion) {
     this.lifeVersion = lifeVersion;
   }
 
+  /**
+   * Constructs a {@link UndeleteResponse} with a valid lifeVersion. The error code will be set to {@link ServerErrorCode#No_Error}.
+   * @param correlationId correlationId of the undelete response.
+   * @param clientId clientId of the undelete response.
+   * @param lifeVersion a valid lifeVersion to return to client.
+   * @param error error code returned in this undelete response.
+   */
+  public UndeleteResponse(int correlationId, String clientId, short lifeVersion, ServerErrorCode error) {
+    super(RequestOrResponseType.UndeleteResponse, UNDELETE_RESPONSE_VERSION_1, correlationId, clientId, error);
```

Comment:
since there is no live site traffic performing undeletes yet, this schema change should be ok, right?

Reply:
that's right.